providers/encoders.inc

```diff
@@ -11,102 +11,161 @@
+            x25519_priv_to_der_encoder_functions),
+    ENCODER("X25519", "yes", "der", "public",
+            x25519_pub_to_der_encoder_functions),
     ENCODER("X25519", "yes", "pem", "private",
-        x25519_priv_pem_encoder_functions),
-    ENCODER("X25519", "yes", "pem", "public", x25519_pub_pem_encoder_functions),
+            x25519_priv_to_pem_encoder_functions),
+    ENCODER("X25519", "yes", "pem", "public",
+            x25519_pub_to_pem_encoder_functions),
 
-    ENCODER("X448", "no", "text", "private", x448_priv_print_encoder_functions),
-    ENCODER("X448", "no", "text", "public", x448_pub_print_encoder_functions),
-    ENCODER("X448", "no", "der", "private", x448_priv_der_encoder_functions),
-    ENCODER("X448", "no", "der", "public", x448_pub_der_encoder_functions),
-    ENCODER("X448", "no", "pem", "private", x448_priv_pem_encoder_functions),
-    ENCODER("X448", "no", "pem", "public", x448_pub_pem_encoder_functions),
+    ENCODER("X448", "no", "text", "private",
+            x448_priv_to_text_encoder_functions),
+    ENCODER("X448", "no", "text", "public",
+            x448_pub_to_text_encoder_functions),
+    ENCODER("X448", "no", "der", "private",
+            x448_priv_to_der_encoder_functions),
+    ENCODER("X448", "no", "der", "public",
+            x448_pub_to_der_encoder_functions),
+    ENCODER("X448", "no", "pem", "private",
+            x448_priv_to_pem_encoder_functions),
+    ENCODER("X448", "no", "pem", "public",
+            x448_pub_to_pem_encoder_functions),
 
     ENCODER("ED25519", "yes", "text", "private",
-        ed25519_priv_print_encoder_functions),
+            ed25519_priv_to_text_encoder_functions),
     ENCODER("ED25519", "yes", "text", "public",
-        ed25519_pub_print_encoder_functions),
+            ed25519_pub_to_text_encoder_functions),
     ENCODER("ED25519", "yes", "der", "private",
-        ed25519_priv_der_encoder_functions),
+            ed25519_priv_to_der_encoder_functions),
     ENCODER("ED25519", "yes", "der", "public",
-        ed25519_pub_der_encoder_functions),
+            ed25519_pub_to_der_encoder_functions),
     ENCODER("ED25519", "yes", "pem", "private",
-        ed25519_priv_pem_encoder_functions),
+            ed25519_priv_to_pem_encoder_functions),
     ENCODER("ED25519", "yes", "pem", "public",
-        ed25519_pub_pem_encoder_functions),
+            ed25519_pub_to_pem_encoder_functions),
 
     ENCODER("ED448", "no", "text", "private",
-        ed448_priv_print_encoder_functions),
-    ENCODER("ED448", "no", "text", "public", ed448_pub_print_encoder_functions),
-    ENCODER("ED448", "no", "der", "private", ed448_priv_der_encoder_functions),
-    ENCODER("ED448", "no", "der", "public", ed448_pub_der_encoder_functions),
-    ENCODER("ED448", "no", "pem", "private", ed448_priv_pem_encoder_functions),
-    ENCODER("ED448", "no", "pem", "public", ed448_pub_pem_encoder_functions),
+            ed448_priv_to_text_encoder_functions),
```

Comment:
same here..

Reply:
Initially ECX was not in the fips provider - but that decision got changed.. So I think they should be yes now.

Reply:
Ok then.  I can change these entries

Reply:
Done